Restore i18n post-UM

ivov opened the pull request:
To close #3015

ivov committed:
:bug: Fix `defaultLocale` watcher

ivov committed:
:zap: Improve error handling for headers

ivov committed:
:pencil2: Improve naming

ivov committed:
:bug: Fix hiring banner check

ivov committed:
:zap: Flatten base text keys

ivov committed:
:zap: Fix miscorrected key

ivov committed:
:zap: Implement pluralization

ivov committed:
:pencil2: Update docs

mutdmour commented on packages/editor-ui/src/App.vue:
```diff
@@ -79,7 +80,7 @@ export default mixins(
 		},
 		logHiringBanner() {
-			if (!this.isHiringBannerEnabled && this.$route.name !== VIEWS.DEMO) {
+			if (this.isHiringBannerEnabled && this.$route.name !== VIEWS.DEMO) {
```
why did you change logic here?

ivov replied:
With the original implementation, `export N8N_HIRING_BANNER_ENABLED=false` does not disable the banner.

mutdmour commented on packages/editor-ui/src/App.vue:
```diff
@@ -160,6 +161,9 @@ export default mixins(
 
 			this.trackPage();
 		},
+		'$store.getters.defaultLocale'(newLocale) {
```
did this not work?
```suggestion
		defaultLocale(newLocale) {
```

ivov replied:
The watcher cannot read from the mapped getter directly:

<img width="1433" alt="image" src="https://user-images.githubusercontent.com/44588767/161282337-8ec95306-2efb-43e1-a989-7e6d5f39a1d1.png">


Either using the full store path as I did, or adding an extra getter just for the watcher.

```js
defaultLocale(): string {
	return this.$store.getters.defaultLocale;
},
```

mutdmour commented on packages/editor-ui/src/plugins/i18n/locales/en.json:
```diff
@@ -1,1093 +1,717 @@
+	"tagsTable.usage": "Usage",
+	"tagsTableHeader.addNew": "Add new",
+	"tagsTableHeader.searchTags": "Search Tags",
+	"tagsView.inUse": "{count} workflow | {count} workflows",
```
love that I can simply search for this now

ivov replied:
:)

mutdmour commented on packages/editor-ui/src/plugins/i18n/docs/ADDENDUM.md:
```diff
@@ -2,27 +2,33 @@
 
 ## Base text
```
thanks for updating the guide. can you also update section that says we should use first words of a message? I think I saw this in a previous PR. Instead names should be more descriptive.

ivov replied:
That section was in [the other PR](https://github.com/n8n-io/n8n/pull/2693/files) that was not merged. I intentionally left that part out.

mutdmour commented on packages/editor-ui/src/views/NodeView.vue:
```diff
@@ -2746,12 +2737,8 @@ export default mixins(
-					} catch (_) {
-						// no headers available
-					}
+					const headers = await this.restApi().getNodeTranslationHeaders();
+					if (headers) addHeaders(headers, this.defaultLocale);
```
should not this be moved to App.vue? otherwise, won't apply if you load other pages

ivov replied:
I moved the headers up because they are used in credentials modal → `Details` → `Allow use by`. Otherwise they wouldn't be fetched if you look at this modal without first going through the canvas.

However, this surfaced an issue that the checkboxes after `Allow use by` are not getting shown if going directly from `/templates` to credentials modal and then to `Details`. I'm seeing this issue in English and non-English locales, so looks unrelated to i18n. Fix here or separately?

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/44588767/161288424-eb74960b-8679-4eb4-8a6c-9cc06bd51ec5.png">
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/44588767/161288498-aa6165b9-d95d-4dc3-934c-40e0c2a033cf.png">

ivov replied:
Templates bug so will be fixed separately.

ivov committed:
:truck: Move headers fetching to `App.vue`

mutdmour committed:
fix hiring banner

ivov committed:
Merge branch 'master' into n8n-3253-restore-i18n-post-um

ivov committed:
:zap: Fix missing import

ivov committed:
:pencil2: Alphabetize translations

ivov committed:
:zap: Switch to async check

ivov commented on packages/cli/src/Server.ts:
```diff
@@ -1503,10 +1503,13 @@ class App {
 					const packagesPath = pathJoin(__dirname, '..', '..', '..');
 					const headersPath = pathJoin(packagesPath, 'nodes-base', 'dist', 'nodes', 'headers');
+
+					if (!existsSync(`${headersPath}.js`)) return;
```
Thanks, switched to promisified `access`, still working:

<img width="191" alt="image" src="https://user-images.githubusercontent.com/44588767/162722404-5cd8046d-9a8c-41fb-8c1f-8e5a06a6435f.png">

janober replied:
Great thanks for the fast fix!

mutdmour committed:
feat(editor): Refactor Output Panel + fix i18n issues (#3097)

* update main panel

* finish up tabs

* fix docs link

* add icon

* update node settings

* clean up settings

* add rename modal

* fix component styles

* fix spacing

* truncate name

* remove mixin

* fix spacing

* fix spacing

* hide docs url

* fix bug

* fix renaming

* refactor tabs out

* refactor execute button

* refactor header

* add more views

* fix error view

* fix workflow rename bug

* rename component

* fix small screen bug

* move items, fix positions

* add hover state

* show selector on empty state

* add empty run state

* fix binary view

* 1 item

* add vjs styles

* show empty row for every item

* refactor tabs

* add branch names

* fix spacing

* fix up spacing

* add run selector

* fix positioning

* clean up

* increase width of selector

* fix up spacing

* fix copy button

* fix branch naming; type issues

* fix docs in custom nodes

* add type

* hide items when run selector is shown

* increase selector size

* add select prepend

* clean up a bit

* Add pagination

* add stale icon

* enable stale data in execution run

* Revert "enable stale data in execution run"

8edb68dbffa0aa0d8189117e1a53381cb2c27608

* move metadata to its own state

* fix smaller size

* add scroll buttons

* update tabs on resize

* update stale data on rename

* remove metadata on delete

* hide x

* change title colors

* binary data classes

* remove duplicate css

* add colors

* delete unused keys

* use event bus

* update styles of pagination

* fix ts issues

* fix ts issues

* use chevron icons

* fix design with download button

* add back to canvas button

* add trigger warning disabled

* show trigger warning tooltip

* update button labels for triggers

* update node output message

* fix add-option bug

* add page selector

* fix pagination selector bug

* fix executions bug

* remove hint

* add json colors

* add colors for json

* add color json keys

* fix select options bug

* update keys

* address comments

* update name limit

* align pencil

* update icon size

* update radio buttons height

* address comments

* fix pencil bug

* change buttons alignment

* fully center

* change order of buttons

* add no output message in branch

* scroll to top

* change active state

* fix page size

* all items

* update expression background

* update naming

* align pencil

* update modal background

* add schedule group

* update schedule nodes messages

* use ellpises for last chars

* fix spacing

* fix tabs issue

* fix too far data bug

* fix executions bug

* fix table wrapping

* fix rename bug

* add padding

* handle unkown errors

* add sticky header

* ignore empty input, trim node name

* nudge lightness of color

* center buttons

* update pagination

* set colors of title

* increase table font, fix alignment

* fix pencil bug

* fix spacing

* use date now

* address pagination issues

* delete unused keys

* update keys sort

* fix prepend

* fix radio button position

* Revert "fix radio button position"

ae42781786f2e6dcfb00d1be770b19a67f533bdf

janober commented:
Got merged with #3097

janober commented:
Thanks a lot. Got merged.
